src/output.cpp

```diff
@@ -2608,20 +2608,18 @@ void print_global_and_character_mode_headers( const catacurses::window &w, const
 
 //generate colorcoded shortcut text
 std::string shortcut_text( const std::string &fmt, keybinding_hint_state state,
-                           const bool shortcut_only )
+                           const bool remove_shortcut_from_text )
 {
     size_t pos = fmt.find_first_of( '<' );
     size_t pos_end = fmt.find_first_of( '>' );
     if( pos_end != std::string::npos && pos < pos_end ) {
         size_t sep = std::min( fmt.find_first_of( '|', pos ), pos_end );
         std::string shortcut = fmt.substr( pos + 1, sep - pos - 1 );
-        if( shortcut_only ) {
-            return shortcut;
-        }
         std::string prestring = fmt.substr( 0, pos );
         std::string poststring = fmt.substr( pos_end + 1, std::string::npos );
 
-        std::string hint_text = string_format( "%s%s%s", prestring, shortcut,
+        std::string hint_text = string_format( "%s%s%s", prestring,
+                                               remove_shortcut_from_text ? "" : shortcut,
                                                poststring );
         return input_context::get_hint_basic( shortcut, hint_text, state );
```

Comment:
> Well, I thought removing the colon would make it replace the first letter with the hotkey even if the string contains spaces but I forgot you reverted that change.

It did do that. The menu looked OK but it used the inlined separators instead of the separate separators ("(a): Blah" instead of "[a]: Blah"). This wasn't consistent so I added this extra `remove_shortcut_from_text ` to the `shortcut_text` function as a way of dealing with this edge case. The code works as expected in every other context the `shortcut_text` function is used (I'm pretty sure) because the `remove_shortcut_from_text` param defaults to false.

For example, this is what the french menu looks like:
![image](https://user-images.githubusercontent.com/2420411/228965118-4394658c-d1ee-4e51-b27f-0b218283ca88.png)

Just to clarify, are there other reasons to implement the solution you've described as opposed to what has already been implemented?

Reply:
Can you test the Japanese menu and let me know if it prints the correct strings? They translated `<W|w> world` as `<W|w> 世界` and I'm concerned it may become `[W] W 世界` with the current code.

Reply:
Good catch, ya this is broken.

![image](https://user-images.githubusercontent.com/2420411/229115655-3c808ed7-4bd9-41b6-bb7f-f5919479a2d7.png)

This change should resolve the issue but I'll need to test this more. https://github.com/CleverRaven/Cataclysm-DDA/pull/64478/commits/394e371de9f1ff2c575251637b6d0b9aef1bae8f

There's a lot of edge cases ;)

Reply:
Please test the Chinese (simplified and traditional) menu with inline format disabled in the settings. They removed the space from the translation (so `<W|w> World` becomes `<W|w>世界` instead of `<W|w> 世界`) and I think the newly added checks will not work in that case.

Although my point is that it's hard to add checks for data that are not (fully) controled by the project itself so I think it may still be more approriate to only replace the hotkey tags with the formatted hotkeys in place and disregard the user's inline setting, at least for now until we fully migrate the main menu hotkey system to be more in line with the rest of the game.